common/src/main/java/com/microsoft/identity/common/internal/controllers/ApiDispatcher.java

```diff
@@ -50,6 +52,117 @@
+        sSilentExecutor.execute(new Runnable() {
+            @Override
+            public void run() {
+                initializeDiagnosticContext();
+
+                List<AccountRecord> result = null;
+                BaseException baseException = null;
+                Handler handler = new Handler(Looper.getMainLooper());
+
+                try {
+                    //Try executing request
+                    result = command.execute();
+
+                } catch (final Exception e) {
+                    //Capture any resulting exception and map to MsalException type
+                    Logger.errorPII(
+                            TAG + methodName,
+                            "Silent request failed with Exception",
+                            e
+                    );
+                    if (e instanceof BaseException) {
+                        baseException = (BaseException) e;
+                    } else {
+                        baseException = ExceptionAdapter.baseExceptionFromException(e);
+                    }
+                }
+
+                if (baseException != null) {
+                    //Post On Error
+                    final BaseException finalException = baseException;
+                    handler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            command.getCallback().onError(finalException);
+                        }
+                    });
+                } else {
+                    final List<AccountRecord> finalAccountsList = result;
+
+                    handler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            command.getCallback().onSuccess(finalAccountsList);
+                        }
+                    });
+                }
+            }
+        });
+    }
+
+    public static void removeAccount(final RemoveAccountCommand command) {
+        final String methodName = ":getAccounts";
+        Logger.verbose(
+                TAG + methodName,
+                "Beginning load accounts."
+        );
+        sSilentExecutor.execute(new Runnable() {
+            @Override
+            public void run() {
+                initializeDiagnosticContext();
+
+                boolean result = false;
+                BaseException baseException = null;
+                Handler handler = new Handler(Looper.getMainLooper());
+
+                try {
+                    //Try executing request
+                    result = command.execute();
+
+                } catch (final Exception e) {
+                    //Capture any resulting exception and map to MsalException type
+                    Logger.errorPII(
+                            TAG + methodName,
+                            "Silent request failed with Exception",
+                            e
+                    );
+                    if (e instanceof BaseException) {
```

Comment:
Same as above. `baseExceptionFromException` should handle this already.... if it doesn't _it should_